### DISCUSSION TITLE
Relationship to https://github.com/eclipse/aspectj.eclipse.jdt.core

### DISCUSSION THREAD

**Discussion by jcrben** [eclipse-jdt/eclipse.jdt.core#589] [Q&A] [answered]
This looks like pretty much the same code as https://github.com/eclipse/aspectj.eclipse.jdt.core - what's the difference? If they are the same why have two duplicate repositories on github?

Currently the README looks the same, but could be updated to clarify the differences

**Comment by iloveeclipse** [comment]
Well, this repo is the main one, and anyone is free to clone & create forks / do whatever is compatible with the project license.
I can't say what aspectj does with jdt core code.

**Reply by jcrben** [reply]
Good to know this is the main one as that wasn't totally clear.

It's not a Github fork tho - if it was, it would come up at https://github.com/eclipse-jdt/eclipse.jdt.core/network/members and be more obvious about the downstream relationship.

I'd expect the Eclipse Foundation to be coordinated some such a core piece. We'll see what they say.

**Comment by kriegaex** [comment, marked as answer]
The relationship is that Eclipse AspectJ uses the compiler and APT parts of JDT Core as a basis for the AspectJ Compiler (AJC) for compile-time weaving and its weaving agent `aspectjweaver.jar` for load-time weaving. If you look the the main branch, of course they look identical, because the AspectJ team does not commit to _main_, but to the [_aspectj_ branch](https://github.com/eclipse/aspectj.eclipse.jdt.core/tree/aspectj) or sub-branches thereof. That makes it easier to refresh the main branch from upstream.

There are also technical details, such as originally both repositories having resided in the same GitHub organisation https://github.com/eclipse, but within the same organisation it was impossible to mark one repository as a fork of the other one. So we had to push a copy. JDT Core recently moved to its own organisation https://github.com/eclipse-jdt, which would probably make it possible for admins to change the fork status of AspectJ's JDT Core repo. The AspectJ project is also in discussions with the admin team to move our repositories into our own _eclipse-aspectj_ organisation, but the project lead is busy and did not get around to actually make a formal request. I think that after the move, we might be able to address the issue of marking our repo as a fork of the upstream JDT Core one and then maybe pull our changes over to the main branch, also adjusting the read-me file in the process.

**Reply by jcrben** [reply]
Yo could possibly set the default branch which displays to something other than the master branch. Just a thought. Thanks for your response!

**Reply by kriegaex** [reply]
@aclement, maybe you as the project lead could change the repo settings and make the _aspectj_ branch the default branch. Actually, that is not such a bad idea.

**Reply by kriegaex** [reply]
Default branch changed, see https://github.com/eclipse-aspectj/eclipse.jdt.core/issues/27#issuecomment-1504848759.
